command/agent/auth/jwt/jwt.go

```diff
@@ -160,8 +163,11 @@ func (j *jwtMethod) ingressToken() {
 
 	j.logger.Debug("new jwt file found")
 
-	if !fi.Mode().IsRegular() {
-		j.logger.Error("jwt file is not a regular file")
+	// Check that the path refers to a file.
+	// If it's a sym link, it could still be a symlink to a directory,
+	// but ioutil.ReadFile below will return a descriptive error.
+	if !fi.Mode().IsRegular() && (fi.Mode()&fs.ModeSymlink) == 0 {
+		j.logger.Error("jwt file is not a regular file or symlink")
 		return
 	}
```

Comment:
I was thinking a switch statement might make the logic clearer here, though now that I've written it out I'm not so sure. Leaving it here for posterity 🙂 

```suggestion
	switch mode := fi.Mode(); {
	case mode.IsRegular():
		// regular file
	case mode&fs.ModeSymlink != 0:
		// symlink
	default:
		j.logger.Error("jwt file is not a regular file or symlink")
		return
	}
```

Reply:
I actually prefer that, thanks for the suggestion.